### PR TITLE
Remextract

### LINKED ISSUE #112: rewrite cmdlinearchextractor to python + fnmap tool not use posix and comment code

### PR DESCRIPTION
**Pull request by dkrupp**
partly fixes #112.  Functionality of the clang-cmd-line-extractor tool is implemented in xtu-build.py script.

### CODE REVIEW

**Comment by gerazo** [comment]
I like it. Shouldn't we put it into ctu-os?

**Comment by dkrupp** [comment]
if we accept it here. I will merge that to ctu-os tomorrow.

**Comment by gerazo** [comment]
It is ok for me. @Xazax-hun ?

File: tools/xtu-build-new/xtu-build.py [hunk 1]
```diff
@@ -119,15 +120,28 @@ def get_command_arguments(cmd):
...
     cmd = src_2_cmd[source]
-    args = get_command_arguments(cmd)
-    arch_command = os.path.join(clang_path, 'clang-cmdline-arch-extractor') + \
-        ' ' + string.join(args, ' ') + ' ' + source
-    if mainargs.verbose:
-        print arch_command
-    arch_output = subprocess.check_output(arch_command, shell=True)
-    arch = arch_output[arch_output.rfind('@')+1:].strip()
+    args = get_command_arguments(cmd)        
+    arch=get_triple_arch(clang_path,args,source)
```
Comment (Xazax-hun) [inline tools/xtu-build-new/xtu-build.py]: Inconsistent spacing across the operators.

File: tools/xtu-build-new/xtu-build.py [hunk 2]
```diff
@@ -119,15 +120,28 @@ def get_command_arguments(cmd):
...
+def get_triple_arch(clang_path, clang_args,source):
+    """Returns the architecture part of the target triple in a compilation command """
+    arch = ""
+    clang_cmd = []
+    clang_cmd.append(os.path.join(clang_path, 'clang'))
+    clang_cmd.append("-###")
+    clang_cmd.extend(clang_args)
+    clang_cmd.append(source)    
+    clang_out = subprocess.check_output(clang_cmd, stderr=subprocess.STDOUT, shell=False)    
+    clang_params=shlex.split(clang_out)
```
Comment (Xazax-hun) [inline tools/xtu-build-new/xtu-build.py]: please add spaces here as well.

File: tools/xtu-build-new/xtu-build.py [hunk 3]
```diff
@@ -119,15 +120,28 @@ def get_command_arguments(cmd):
...
+    i=0
+    while i<len(clang_params) and clang_params[i]!="-triple":        
+        i=i+1
+    if i<(len(clang_params) - 1):
```
Comment (Xazax-hun) [inline tools/xtu-build-new/xtu-build.py]: And here.

**Pull request merged**